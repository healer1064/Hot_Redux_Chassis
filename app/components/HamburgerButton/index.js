/** @flow */

import styles from './HamburgerButton.css';
import React from 'react';
import cx from 'classnames';

function HamburgerButton({ open, ...props }) {
  return (
    <div
      className={cx(
        styles.hamburgerButton,
        open && styles.open
      )}
      aria-role="button"
      {...props}
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  );
}

export default HamburgerButton;
