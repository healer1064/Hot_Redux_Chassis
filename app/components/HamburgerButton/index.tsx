import React, { HTMLProps } from 'react';
import styles from './HamburgerButton.css';
import cx from 'classnames';

type Props = {
  open: boolean;
} & HTMLProps<HTMLButtonElement>;

function HamburgerButton({ open, ...props }: Props) {
  return (
    <button
      className={cx(styles.hamburgerButton, open === true && styles.open)}
      {...(props as any)}
    >
      <div />
      <div />
      <div />
      <div />
    </button>
  );
}

export default HamburgerButton;
