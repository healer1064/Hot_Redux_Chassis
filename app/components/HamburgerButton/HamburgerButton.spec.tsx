import HamburgerButton from './';
import { testComponentSnapshots } from 'tests/helpers';

describe('<HamburgerButton />', () => {
  testComponentSnapshots(HamburgerButton, [{ open: false }, { open: true }]);
});
