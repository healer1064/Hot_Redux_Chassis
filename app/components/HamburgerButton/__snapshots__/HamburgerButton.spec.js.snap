// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HamburgerButton /> should match snapshot 1`] = `
<div
  className="hamburgerButton"
>
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`<HamburgerButton /> should match snapshot 2`] = `
<div
  className="hamburgerButton open"
>
  <div />
  <div />
  <div />
  <div />
</div>
`;
